### ISSUE TITLE
Natural Sort Plugin is Case Sensitive

### ISSUE THREAD

**Issue by freak3dot** [DataTables/Plugins#281]
The natural sort plugin references Jim Palmers code which included an option for case insensitive.
http://www.overset.com/2008/09/01/javascript-natural-sort-algorithm-with-unicode-support/

`naturalSort.insensitive = true;`

I'm thinking the plugin should have options for case sensitive and insensitive. I was thinking we should add two new sort types.

```
jQuery.extend( jQuery.fn.dataTableExt.oSort, {

        [...]

	"natural-ci-asc": function ( a, b ) {
		return naturalSort(a.toString().toLowerCase(),b.toString().toLowerCase());
	},

	"natural-ci-desc": function ( a, b ) {
		return naturalSort(a.toString().toLowerCase(),b.toString().toLowerCase()) * -1;
	}
} );
```

The goal here would be to be able to mix natural case sensitive and natural case insensitive in the same table.

I'd be happy to update the code. However, I wanted to get thoughts from those involved in the project first.

**Comment by DataTables** [comment]
Sounds very good to me.

I have some slight concern about the fact that the `insensitive` parameter is static, but I think it would simply mean that you would simply need to set that parameter every time immediately before calling the `naturalSort` method, since it could be multi-column sorting over natural sorting case-insensitive and case-sensitive columns.

The other option would be to use `toLowerCase()`, that could even be done in a `-pre` function which means it would only be done once for each data point.

**Comment by DataTables** [comment]
Thanks for the PR to fix this one :-)

**Comment by freak3dot** [comment]
I noticed something weird with the pre function on this. It wasn't sorting numeric table columns naturally in our application. If I removed the pre function and did the toLowerCase in the 'natural-ci-asc' and 'natural-ci-desc' I got the expected results.

I noticed the pre function was referred to as a formatter in the DataTables source. Is the pre function doing a sort of it's own?

In case it matters, I'm using version 1.10.12.

**Comment by DataTables** [comment]
Ah! Yes, that's a bug - although its a grey area if its in DataTables or the plug-in... If there is a `pre` function, DataTables assumes that any formatting needed is done there and there is no need for a custom `asc` or `desc` method. Your workaround is a good one and I'll commit that in.

**Comment by DataTables** [comment]
Fixed in 815ca8e59da011ebf9e451c2f221c59b893a8085. Thanks for bringing that up!
